### PR TITLE
feat(onyx-637): Add PartnerOfferCreated activity event

### PR DESCRIPTION
**Pull request by nickskalkin**
[Jira ticket](https://artsyproduct.atlassian.net/browse/ONYX-637)
[Gravity PR](https://github.com/artsy/gravity/pull/17224)

Adds support for such queries:

```graphql
query MyQuery {
  me {
    email
    notification(id: "...") {
      title
      message
      
      item {
        ... on PartnerOfferCreatedNotificationItem {
          expiresAt
          
          artworksConnection(first: 1) {
            edges {
              node {
                title
              }
            }
          }
        }
      }
    }
  }
}
```

```graphql
query MyQuery{ 
  notificationsConnection(first: 5) {
    edges {
      node {
        title
        message
        
        item {
          ... on PartnerOfferCreatedNotificationItem {
            expiresAt
            
            artworksConnection(first: 1) {
              edges {
                node {
                  title
                }
              }
            }
          }
        }
      }
    }
  }
}
```

### CODE REVIEW

File: src/schema/v2/notifications/Item/__tests__/index.test.ts [hunk 1]
```diff
@@ -431,4 +433,85 @@ describe("NotificationItem", () => {
       `)
     })
   })
+
+  describe('for "PartnerOfferCreatedNotificationItem"', () => {
+    const artworksLoader = jest.fn(() =>
+      Promise.resolve([
+        {
+          id: "artwork-id",
+          title: "Catty Artwork",
```
Comment (nickskalkin) [inline src/schema/v2/notifications/Item/__tests__/index.test.ts:442]: 🐈‍⬛

File: src/schema/v2/notifications/Item/PartnerOfferCreatedNotificationItem.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      resolve: async (
+        { object_ids },
+        args,
+        { artworksLoader, mePartnerOfferLoader }
+      ) => {
+        if (!mePartnerOfferLoader) return null
+        // object_ids should include a single partner offer id
+        if (object_ids.length === 0) return null
+
+        const { artwork_id } = await mePartnerOfferLoader(object_ids[0])
```
Comment (mzikherman) [inline src/schema/v2/notifications/Item/PartnerOfferCreatedNotificationItem.ts:27]: Is this right? I think `object_ids` are the artwork id's already, while offer, which isn't needed to resolve this connection, are the `actor_ids`. So you can use `object_ids` directly and don't need this loader here either.

File: src/schema/v2/notifications/Item/PartnerOfferCreatedNotificationItem.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        return {
+          totalCount,
+          pageCursors: createPageCursors({ page, size }, totalCount),
+          ...connectionFromArray(body, args),
+        }
+      },
+    },
+    expiresAt: {
+      type: GraphQLString,
```
Comment (mzikherman) [inline src/schema/v2/notifications/Item/PartnerOfferCreatedNotificationItem.ts:41]: We have a more structured type for timestamps, that support some common client formatting needs - https://github.com/artsy/metaphysics/blob/98e7b3b7f6dc00932f9614b87f885417d51a4a08/src/lib/date.ts - should likely use that.

File: src/schema/v2/notifications/Item/PartnerOfferCreatedNotificationItem.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      resolve: async ({ object_ids }, _, { mePartnerOfferLoader }) => {
+        if (!mePartnerOfferLoader) return null
+        // object_ids should include a single partner offer id
```
Comment (mzikherman) [inline src/schema/v2/notifications/Item/PartnerOfferCreatedNotificationItem.ts:44]: Same as above, I think `object_ids` and `actor_ids` are getting swapped around here vs. the Gravity PR.

File: src/schema/v2/notifications/index.ts [hunk 5]
```diff
@@ -104,7 +105,20 @@ export const NotificationType = new GraphQLObjectType<any, ResolverContext>({
     artworksConnection: {
       type: artworkConnection.connectionType,
       args: pageable(),
-      resolve: async ({ object_ids: ids }, args, { artworksLoader }) => {
+      resolve: async (
+        { notificationType, object_ids: ids },
+        args,
+        { artworksLoader, mePartnerOfferLoader }
+      ) => {
+        if (notificationType == "PartnerOfferCreatedActivity") {
```
Comment (mzikherman) [inline src/schema/v2/notifications/index.ts:113]: As above, I _think_ this isn't needed as `object_ids` should already contain the artwork id's.

**Review by mzikherman — CHANGES_REQUESTED** [review]
Looks good overall! Main blocking comment(s) are about the `object_ids` vs `actor_ids` confusion and which one contains the artwork id's.

**Comment by nickskalkin** [comment]
@mzikherman sorry for the confusion between actors / objects, I've explained the reason here: https://github.com/artsy/gravity/pull/17224#issuecomment-1882772252

I did 360 and now objects are artworks, which simplifies metaphysics code a bit, could you re-review?

**Review by mzikherman — APPROVED** [review]
Nice!

**Pull request merged**